### ISSUE TITLE
Integration issue with goconst in v2.12.1

### ISSUE THREAD

**Issue by YakDriver** [golangci/golangci-lint#6563]
### Welcome

- [x] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [x] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [x] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/docs/welcome/faq/#why-do-you-have-typecheck-errors).
- [x] Yes, I've tried with the standalone [linter](https://golangci-lint.run/docs/linters/) if available (e.g., gocritic, go vet, etc.).
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### How did you install golangci-lint?

go install

### Description of the problem

`goconst` counts string literal occurrences at the package level across all files (including excluded files), then reports that global count in non-excluded files. This is new going from [golangci-lint v2.11.4 / goconst v1.8.2 to golangci-lint v2.12.1 / goconst v1.10.0](https://github.com/hashicorp/terraform-provider-aws/actions/runs/25304556562/job/74177759021?pr=47740).

For example, a string appearing once in `bool.go` and 17 times in `bool_test.go` is reported as "18 occurrences" in `bool.go`, even when test files are excluded from reporting. 

If we exclude test files in golangci-lint config:

```yml
version: "2"
linters:
  default: none
  enable:
    - goconst
  settings:
    goconst:
      min-occurrences: 4
  exclusions:
    rules:
      - path: _test\.go
        linters:
          - goconst
```

We still get a finding:

```
../internal/sdkv2/types/nullable/bool.go:45:19: string `false` has 18 occurrences, make it a constant (goconst)
	case "", "true", "false":
	                 ^
```

### References

* [Terraform AWS Provider failing CI](https://github.com/hashicorp/terraform-provider-aws/actions/runs/25304556562/job/74177759021?pr=47740)
* jgautheron/goconst#57

### Version of golangci-lint

golangci-lint has version 2.12.1 built with go1.26.2 from (unknown, modified: ?, mod sum: "h1:Rr07Ps/u+kvQqSO4L026WOSVvOlMU+RZTa/lossFaJA=") on (unknown)

### Configuration

<details>

```console
version: "2"
linters:
  default: none
  enable:
    - goconst
  settings:
    goconst:
      min-occurrences: 4
  exclusions:
    rules:
      - path: _test\.go
        linters:
          - goconst
```

</details>

### Go environment

<details>

```console
go version go1.26.2 darwin/arm64
AR='ar'
CC='/opt/homebrew/opt/llvm/bin/clang'
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_ENABLED='1'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
CXX='/opt/homebrew/opt/llvm/bin/clang++'
GCCGO='gccgo'
GO111MODULE=''
GOARCH='arm64'
GOARM64='v8.0'
GOAUTH='netrc'
GOBIN='/develop/go/bin'
GOCACHE='/tmp/go-cache'
GOCACHEPROG=''
GODEBUG=''
GOENV='/Users/oshkoskbgosh/Library/Application Support/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFIPS140='off'
GOFLAGS=''
GOGCCFLAGS='-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -ffile-prefix-map=/tmp/go-tmp/go-build2082944235=/tmp/go-build -gno-record-gcc-switches -fno-common'
GOHOSTARCH='arm64'
GOHOSTOS='darwin'
GOINSECURE=''
GOMOD='/develop/go/src/github.com/hashicorp/terraform-provider-aws/go.mod'
GOMODCACHE='/tmp/go-mod-cache'
GONOPROXY=''
GONOSUMDB=''
GOOS='darwin'
GOPATH='/develop/go'
GOPRIVATE=''
GOPROXY='https://proxy.golang.org,direct'
GOROOT='/opt/homebrew/Cellar/go/1.26.2/libexec'
GOSUMDB='sum.golang.org'
GOTELEMETRY='local'
GOTELEMETRYDIR='/Users/oshkoskbgosh/Library/Application Support/go/telemetry'
GOTMPDIR='/tmp/go-tmp'
GOTOOLCHAIN='auto'
GOTOOLDIR='/opt/homebrew/Cellar/go/1.26.2/libexec/pkg/tool/darwin_arm64'
GOVCS=''
GOVERSION='go1.26.2'
GOWORK=''
PKG_CONFIG='pkg-config'
```

</details>

### Verbose output of running

<details>

```console
% golangci-lint run --config ../../../../.ci/constants.yml --verbose
INFO golangci-lint has version 2.12.1 built with go1.26.2 from (unknown, modified: ?, mod sum: "h1:Rr07Ps/u+kvQqSO4L026WOSVvOlMU+RZTa/lossFaJA=") on (unknown) 
INFO [config_reader] Used config file ../../../../.ci/constants.yml 
INFO [config_reader] Module name "github.com/hashicorp/terraform-provider-aws" 
INFO [goenv] Read go env for 9.894667ms: map[string]string{"GOCACHE":"/tmp/go-cache", "GOROOT":"/opt/homebrew/Cellar/go/1.26.2/libexec"} 
INFO [lintersdb] Active 1 linters: [goconst]      
INFO [loader] Go packages loading at mode 8767 (files|imports|name|compiled_files|deps|exports_file|types_sizes) took 331.583125ms 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 1.003583ms 
INFO [linters_context/goanalysis] analyzers took 0s with no stages 
INFO [runner/exclusion_rules] Skipped 2 issues by rules: [Path: "_test\\.go", Linters: "goconst"] 
INFO [runner] Issues before processing: 4, after processing: 1 
INFO [runner] Processors filtering stat (in/out): uniq_by_line: 2/1, max_from_linter: 1/1, path_absoluter: 4/4, filename_unadjuster: 4/4, exclusion_rules: 4/2, diff: 2/2, max_same_issues: 1/1, severity-rules: 1/1, cgo: 4/4, path_relativity: 4/4, exclusion_paths: 4/4, generated_file_filter: 4/4, sort_results: 1/1, invalid_issue: 4/4, fixer: 2/2, max_per_file_from_linter: 1/1, source_code: 1/1, path_shortener: 1/1, path_prettifier: 1/1, nolint_filter: 2/2 
INFO [runner] processing took 755.832µs with stages: nolint_filter: 620.917µs, generated_file_filter: 85.707µs, source_code: 25.291µs, exclusion_rules: 11.458µs, path_relativity: 6.25µs, max_same_issues: 1.209µs, uniq_by_line: 1.167µs, cgo: 583ns, invalid_issue: 583ns, sort_results: 499ns, path_shortener: 459ns, max_from_linter: 458ns, fixer: 250ns, path_prettifier: 209ns, exclusion_paths: 208ns, path_absoluter: 208ns, max_per_file_from_linter: 167ns, filename_unadjuster: 125ns, diff: 42ns, severity-rules: 42ns 
INFO [runner] linters took 288.1765ms with stages: goanalysis_metalinter: 287.367167ms 
../internal/sdkv2/types/nullable/bool.go:45:19: string `false` has 18 occurrences, make it a constant (goconst)
	case "", "true", "false":
	                 ^
1 issues:
* goconst: 1
INFO File cache stats: 1 entries of total size 1.8KiB 
INFO Memory: 8 samples, avg is 51.9MB, max is 65.7MB 
INFO Execution took 631.344417ms 
```

</details>

### A minimal reproducible example or link to a public repository

<details>

[bool.go](https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/sdkv2/types/nullable/bool.go#L45) - 1 occurence of "false"
[bool_test.go](https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/sdkv2/types/nullable/bool_test.go) - 17 occurences of "false"

</details>

### Validation

- [x] Yes, I've included all the information above (example, version, config, etc.).

### Supporter

- [ ] I am a [sponsor/backer](https://donate.golangci.org) of this project.

**Comment by ldez** [comment]
Due to the very specific way `goconst` works, we added a new option in v2.12.0:

```yml
linters:
  settings:
    goconst:

      # Ignore strings from test files.
      # Default: false
      ignore-tests: true
```


https://golangci-lint.run/docs/linters/configuration/#goconst

**Comment by YakDriver** [comment]
Works great! Thank you

**Comment by timc13** [comment]
that works specifically for `_test\.go` but what if I have some mocks in a `mocks/` folder I would like to ignore?

**Comment by ldez** [comment]
goconst it accumulates the constants from a package.

By using golangci-lint exclusions you can ignore a package, but this is not possible for a specific files like test files.

So you can use something like that, if you want to exclude `mocks` directory/package:

```yml
version: "2"
linters:
  # ...
  exclusions:
    rules:
      - path: .+/mocks/.+
        linters:
          - goconst
```

Obviously, you must adapt the regular expression for your needs.

Your question is a bit off topic.
